src/libasr/pass/python_bind.cpp

```diff
@@ -0,0 +1,406 @@
+                            ASR::abiType::BindC, ASR::Public, ASR::presenceType::Required, false);
+    ASR::expr_t *pReturn_ref = ASRUtils::EXPR(ASR::make_Var_t(al, f.base.base.loc,
+                                    ASR::down_cast<ASR::symbol_t>(pReturn)));
+    f.m_symtab->add_symbol(std::string("pReturn"), ASR::down_cast<ASR::symbol_t>(pReturn));
+    body.push_back(al, ASRUtils::STMT(
+        ASR::make_Assignment_t(al, f.base.base.loc, pReturn_ref, 
+            ASRUtils::EXPR(ASRUtils::make_FunctionCall_t_util(al, f.base.base.loc, sym_PyObject_CallObject, nullptr,
+                                args_PyObject_CallObject.p, args_PyObject_CallObject.n, ptr_t, nullptr, nullptr)),
+                            nullptr)));
+
+    // Converting CPython result to native type
+    ASR::ttype_t *ret_type = ASRUtils::get_FunctionType(f)->m_return_var_type;
+    if (ret_type) {
+        std::string return_var_name = "_lpython_return_variable";
+        ASR::asr_t *ret_var = (ASR::asr_t*)f.m_symtab->get_symbol(return_var_name);
+        LCOMPILERS_ASSERT(ret_var);
+        ASR::expr_t *ret_var_ref = ASRUtils::EXPR(ASR::make_Var_t(al, f.base.base.loc,
+                                        ASR::down_cast<ASR::symbol_t>(ret_var)));
+        ASR::expr_t *ret_conv_result = cpython_to_native(al, pReturn_ref, ret_type, f, parent_scope);
+        body.push_back(al, ASRUtils::STMT(ASR::make_Assignment_t(al, f.base.base.loc, ret_var_ref, ret_conv_result,
+                                            nullptr)));
+    }
+
+    /*
+        Py_DecRef(pName);
+        Py_DecRef(pArgs);
+        Py_DecRef(pReturn);
+    */
+    ASR::symbol_t *sym_Py_DecRef = parent_scope.get_symbol("Py_DecRef");
+
+    Vec<ASR::call_arg_t> args_Py_DecRef;
+    args_Py_DecRef.reserve(al, 1);                                
+    args_Py_DecRef.push_back(al, {f.base.base.loc, pName_ref});
+    body.push_back(al, ASRUtils::STMT(ASRUtils::make_SubroutineCall_t_util(al, f.base.base.loc, sym_Py_DecRef, nullptr,
+                        args_Py_DecRef.p, args_Py_DecRef.n, nullptr, nullptr, false, false)));
+
+    Vec<ASR::call_arg_t> args_Py_DecRef2;
+    args_Py_DecRef2.reserve(al, 1);
+    args_Py_DecRef2.push_back(al, {f.base.base.loc, pArgs_ref});
+    body.push_back(al, ASRUtils::STMT(ASRUtils::make_SubroutineCall_t_util(al, f.base.base.loc, sym_Py_DecRef, nullptr,
+                        args_Py_DecRef2.p, args_Py_DecRef2.n, nullptr, nullptr, false, false)));
+
+    Vec<ASR::call_arg_t> args_Py_DecRef3;
+    args_Py_DecRef3.reserve(al, 1);
+    args_Py_DecRef3.push_back(al, {f.base.base.loc, pReturn_ref});
+    body.push_back(al, ASRUtils::STMT(ASRUtils::make_SubroutineCall_t_util(al, f.base.base.loc, sym_Py_DecRef, nullptr,
+                        args_Py_DecRef3.p, args_Py_DecRef3.n, nullptr, nullptr, false, false)));
+
+    // reassignment
+    f.m_body = body.p;
+    f.n_body = body.n;
+    ASRUtils::get_FunctionType(f)->m_abi = ASR::abiType::BindC;
```

Comment:
Any reason to make the `abi` as `BindC`? I think we should be making it `Source`?

Reply:
Ok. Will change it to Source in the next commit.